### PR TITLE
Use TypeVar to help with QObject.findChild[ren]()

### PR DESCRIPTION
**Pull request by altendky**

### CODE REVIEW

**Comment by altendky** [comment]
I'm looking into the ones that take tuples of types, but if you know how to handle it...

**Comment by altendky** [comment]
This is disgusting but I don't know of a better way yet.  Maybe it's worth skipping the tuple-taking overloads.

https://repl.it/@altendky/CalmSickDistributeddatabase-1
```python
import string


upper_letters = ['T' + c for c in string.ascii_uppercase]


def create_typevars(
    names=upper_letters,
    typevar='typing.TypeVar',
    bound=None,
):
    if bound is None:
        bound_argument = ''
    else:
        bound_argument = f', bound={bound}'
    return [
        f'{name} = {typevar}("{name}"{bound_argument})'
        for name in names
    ]


def main():
    names = upper_letters[:4]
    print('\n'.join(create_typevars(names=names, bound='QObject')))

    template = 'def findChildren(self, types: typing.Tuple[{0}], name: str = ..., options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> typing.List[typing.Union[{0}]]: ...'

    for index in range(len(names)):
        print(template.format(', '.join(names[:index + 1])))

main()
```

```
TA = typing.TypeVar("TA", bound=QObject)
TB = typing.TypeVar("TB", bound=QObject)
TC = typing.TypeVar("TC", bound=QObject)
TD = typing.TypeVar("TD", bound=QObject)
def findChildren(self, types: typing.Tuple[TA], name: str = ..., options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> typing.List[typing.Union[TA]]: ...
def findChildren(self, types: typing.Tuple[TA, TB], name: str = ..., options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> typing.List[typing.Union[TA, TB]]: ...
def findChildren(self, types: typing.Tuple[TA, TB, TC], name: str = ..., options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> typing.List[typing.Union[TA, TB, TC]]: ...
def findChildren(self, types: typing.Tuple[TA, TB, TC, TD], name: str = ..., options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> typing.List[typing.Union[TA, TB, TC, TD]]: ...
```

**Comment by altendky** [comment]
Kick CI

**Comment by altendky** [comment]
ah, just travis trouble
image

**Comment by altendky** [comment]
Maybe I'll come back and do that fancy generated variadic generics stuff but...  I think it is at least good to keep that separate regardless.

File: PyQt5-stubs/QtCore.pyi [hunk 1]
```diff
@@ -1801,21 +1804,21 @@ class QObject(sip.wrapper):
...
     @typing.overload
-    def findChildren(self, types: typing.Tuple, re: 'QRegularExpression', options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> typing.List['QObject']: ...
+    def findChildren(self, types: typing.Tuple[typing.Type[QObject], ...], re: 'QRegularExpression', options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> typing.List['QObject']: ...
     @typing.overload
-    def findChild(self, type: type, name: str = ..., options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> 'QObject': ...
+    def findChild(self, type: typing.Type[QObjectT], name: str = ..., options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> typing.Optional[QObjectT]: ...
     @typing.overload
-    def findChild(self, types: typing.Tuple, name: str = ..., options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> 'QObject': ...
+    def findChild(self, types: typing.Tuple[typing.Type[QObject], ...], name: str = ..., options: typing.Union[Qt.FindChildOptions, Qt.FindChildOption] = ...) -> 'typing.Optional[QObject]': ...
```
Comment (BryceBeagle) [inline PyQt5-stubs/QtCore.pyi:1821]: Should this (and the other couple Tuple ones) really be expressed as homogeneous tuples?

Comment (BryceBeagle) [reply PyQt5-stubs/QtCore.pyi:1821]: Or is this just the best we can do without variadic generics?

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

File: PyQt5-stubs/QtCore.pyi (hunk 1 above)
Comment (BryceBeagle) [reply PyQt5-stubs/QtCore.pyi:1821]: Scratch that. Forgot this was the thread where this was discussed

**Review by BryceBeagle — APPROVED** [review]

File: PyQt5-stubs/QtCore.pyi (hunk 1 above)
Comment (altendky) [reply PyQt5-stubs/QtCore.pyi:1821]: They should be homogeneous to the degree described here, I think.  And yeah, no new ideas for the variadic generics yet.

**Pull request merged**